### PR TITLE
added section on FlowLayout support

### PR DESCRIPTION
**Pull request by offthegrass**

### CODE REVIEW

**Comment by mitar** [comment]
@rclai said in #19:

> One thing I noticed when I used Flow Layout with BC is that the components would be rendered separate from each other and have no access to each other unless you use getComponentForElement or getView. I tried to render a component to be the layout in order for the region components to inherit from it but it was not possible. Has anyone tried doing this or found a solution?

**Comment by mitar** [comment]
So, does this work in addressing the concerns raised by @rclai? So does this make a hierarchy of components correct?

@offthegrass: Could you also create tests? Create a set of files like `flowlayout.coffee` and `flowlayout.html` in the `tests` directory and create a small example and then you can use a similar approach I use [to dump the whole tree](https://github.com/peerlibrary/meteor-blaze-components/blob/master/tests/tests.coffee#L1984) and see if it has the correct structure. So that the parents are linked up correctly.

**Comment by mitar** [comment]
It is OK to depend on flow layout for tests.

**Comment by mitar** [comment]
BTW, I have not used flow router. Does support for flow layout also gives us support for flow router?

**Comment by mitar** [comment]
Also, I have all examples from the documentation also in tests. To make sure they really works. So maybe you can use example as a test. But it will be hard because of the login/logout. Maybe use a `Session` or something you control.

**Comment by rclai** [comment]
Flow Router and Flow Layout are separate. We only care about Flow Layout.

**Comment by mitar** [comment]
Because Flow Router uses `FlowLayout.render` to its action? So if we handle that to work, then everything is set? Great!

**Comment by rclai** [comment]
I would post a MeteorPad, but it's down for some reason -_-.

**Comment by rclai** [comment]
Just in case you didn't understand, Flow Router is not dependent on Flow Layout because you can do whatever you want in Flow Router's `action`.

We only care about being able to render Blaze Components with `FlowLayout.render('ParentComponent', { region: 'ChildComponent' })`, making sure that the `ParentComponent` actually works as a Blaze Component.

**Comment by mitar** [comment]
Yea. I got that.

File: README.md [hunk 1]
```diff
@@ -797,7 +798,61 @@ for property, value of BlazeComponent when property not in ['__super__']
...
+  {{> FlowLayoutRoot}}
+</body>
+````
+Step 4: Tell FlowLayout to use our **root DOM node**
+```coffee
+FlowLayout.setRoot '#fl-root'
+```
+Now you can use FlowLayout as normal. 
+
+### Example ###
```
Comment (mitar) [inline README.md:837]: This is not the best example. Let's try one with components, not templates. :-)

**Comment by rclai** [comment]
I think this is the line of code in Flow Layout that handles the layout:
https://github.com/meteorhacks/flow-layout/blob/master/lib/client/layout.js#L19
I don't know what needs to be done there. I don't know too much about the deep workings of SpaceBars/Blaze.

**Comment by rclai** [comment]
Should we perhaps get @arunoda's help here?

**Comment by mitar** [comment]
I am not sure what exactly is the current problem? I thought that it works now for you?

**Comment by rclai** [comment]
The problem I'm talking about is this: https://github.com/peerlibrary/meteor-blaze-components/pull/52#commitcomment-11230500 

> Only issue I had was that I couldn't get my layout to function as a component.

File: README.md (hunk 1 above)
Comment (offthegrass) [inline README.md:837]: yeah, this was supposed to be with components. I actually wanted to do a meteorpad with an example, but it was down so I just put this in quickly.

**Comment by offthegrass** [comment]
I did tests with and without a `RootComponent`:
- The `RootComponent` doesn't seem to make any difference, so the above instructions are redundant. My mistake.
- I can confirm the issue @rclai encountered. The components are rendered, but they can't see each other without using `getComponentForElement`, e.g. `@componentParent()` always returns null.

Instead of spending more time on FlowLayout, I made a [LayoutComponent with the same functionality.](https://gist.github.com/offthegrass/3c1491cc9d5b4c5cb5d0). 

Mistakes?

Name ideas?

**Comment by mitar** [comment]
It would be easier for you if you would do `new ReactiveVar '', (a, b) -> a is b`. Then you can just rely on reactivity to rerender when a component is changed.

**Comment by offthegrass** [comment]
for regions? or layout?

**Comment by mitar** [comment]
I saw [this one](https://gist.github.com/offthegrass/3c1491cc9d5b4c5cb5d0#file-component-coffee-L7).

**Comment by offthegrass** [comment]
thanks, I didn't know about that. nice!

**Comment by rclai** [comment]
This is interesting, that approach does away with Flow Layout rendering wow.

**Comment by mitar** [comment]
BTW, why you are not using reactive dict [here](https://gist.github.com/offthegrass/3c1491cc9d5b4c5cb5d0#file-component-coffee-L9)?

Also, now looking at this code, this looks very similar to our `componentChildren` code. Maybe this is an interesting idea. That one way for `componentChildren` to be set is that you render a component. But another could be that if you modify `componentChildren` yourself, then Blaze Components add it to the rendered children. So then this layout `render` method would just add component to `componentChildren`. And Blaze Components would add it to rendered output.

**Comment by mitar** [comment]
(But we would need some way to tell where to render it to. Hm.)

**Comment by offthegrass** [comment]
> BTW, why you are not using reactive dict here?

it was just easier when I was testing. changed now.

**Comment by mitar** [comment]
@offthegrass: Have you made Layout component  into a package?

**Comment by rhyslbw** [comment]
@DominikGuzei, keep an eye on this

**Comment by mitar** [comment]
This pull request is closed, but see #79.

**Pull request closed without merging**